src/saveload/company_sl.cpp

```diff
@@ -497,7 +497,6 @@ static void Check_PLYR()
 	int index;
 	while ((index = SlIterateArray()) != -1) {
 		CompanyProperties *cprops = new CompanyProperties();
-		memset(cprops, 0, sizeof(*cprops));
```

Comment:
Does the constructor initialise everything?

The code must not rely on SaveLoad reading everything, since that will break badly (read: annoying desyncs) when stuff is only loaded for certain savegame versions.

Reply:
I'll amend the constructor

Reply:
Constructing an object is supposed to also automatically initialize all members of the object, either default-initialization (zero fill for POD types) or calling the default constructor on user-defined types. There shouldn't be any situation where memory of an object allocated with `new` remains uninitialized.